ComponentKit/Utilities/CKComponentGestureActions.mm

```diff
@@ -113,26 +113,29 @@ CKComponentViewAttributeValue CKComponentGestureAttribute(Class gestureRecognize
       @YES  // Bogus value, we don't use it.
     };
   }
-
+  
   static auto *reusePoolMap = new std::unordered_map<CKGestureRecognizerReusePoolMapKey, CKGestureRecognizerReusePool *>();
   static CK::StaticMutex reusePoolMapMutex = CK_MUTEX_INITIALIZER;
   CK::StaticMutexLocker l(reusePoolMapMutex);
   auto &reusePool = (*reusePoolMap)[{gestureRecognizerClass, setupFunction}];
   if (reusePool == nullptr) {
     reusePool = new CKGestureRecognizerReusePool(gestureRecognizerClass, setupFunction);
   }
+  // We're capturing the C++ struct in a block. We use the __block storage class which will call the copy constructor
+  // on our struct when the block is copied to the heap.
+  __block CKTypedComponentAction<UIGestureRecognizer *> blockAction = action;
```

Comment:
This is a bummer with the conversion to C++ struct. We don't get implicit block capture anymore, and I don't know if there's a way to ban the implicit capture that will result in reading garbage memory. Finding these bugs will be difficult in the main app, but we probably have to do it...